backend/apps/project_health/views.py

```
@@ -0,0 +1,78 @@
+        existing = RepoHealthScore.objects.filter(repo_url=repo_url).first()
+        if existing:
+            ttl = timedelta(hours=RepoHealthScore.CACHE_TTL_HOURS)
+            if datetime.now(tz=timezone.utc) - existing.updated_at < ttl:
+                return Response(RepoHealthScoreSerializer(existing).data)
+
+        # Run fresh analysis
+        try:
+            metrics = analyze_repository(repo_url)
+        except ValueError as e:
+            return Response({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)